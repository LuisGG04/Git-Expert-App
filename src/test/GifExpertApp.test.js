import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Testeo del componente GifExpertApp', () => {

    test('Debe mostrar una snapshot correcta', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista con las categorías', () => {
        const categories = ["Warzone", "Pokemón"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});