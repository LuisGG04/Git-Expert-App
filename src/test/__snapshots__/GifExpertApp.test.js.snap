// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeo del componente GifExpertApp Debe de mostrar una lista con las categorías 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Warzone"
      key="Warzone"
    />
    <GifGrid
      category="Pokemón"
      key="Pokemón"
    />
  </ol>
</Fragment>
`;

exports[`Testeo del componente GifExpertApp Debe mostrar una snapshot correcta 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
